src/mdns/mdns_mdnssd.cpp

```diff
@@ -830,12 +842,12 @@ void Publisher::Destroy(Publisher *aPublisher)
     delete static_cast<PublisherMDnsSd *>(aPublisher);
 }
 
-void PublisherMDnsSd::Subscription::Release(void)
+void PublisherMDnsSd::ServiceRef::Release(void)
```

Comment:
Looks like `Release` and `DeallocateServiceRef` are doing the same thing. Is it necessary to keep both?

Reply:
May not necessary. But it's beyond the scope of this PR.

We can refactor it later.